crabs/detection_tracking/train_model.py

```diff
@@ -161,12 +164,58 @@ def core_training(self) -> lightning.Trainer:
             self.seed_n,
         )
 
+        # Get checkpoint type
+        if self.checkpoint_path and os.path.exists(self.checkpoint_path):
+            checkpoint = torch.load(self.checkpoint_path)
+            if all(
+                [
+                    param in checkpoint
+                    for param in ["optimizer_states", "lr_schedulers"]
+                ]
+            ):
+                checkpoint_type = "full"  # for resuming training
+                logging.info(
+                    f"Resuming training from checkpoint at: {self.checkpoint_path}"
+                )
+            else:
+                checkpoint_type = "weights"  # for fine tuning
+                logging.info(
+                    f"Fine-tuning training from checkpoint at: {self.checkpoint_path}"
+                )
+        else:
+            checkpoint_type = None
+
         # Get model
-        lightning_model = FasterRCNN(self.config)
+        if checkpoint_type == "weights":
+            # Note: weights-only checkpoint contains hyperparameters
+            # see https://lightning.ai/docs/pytorch/stable/common/checkpointing_basic.html#save-hyperparameters
+            lightning_model = FasterRCNN.load_from_checkpoint(
+                self.checkpoint_path,
+                config=self.config,
+                # overwrite checkpoint hyperparameters with config ones
+                # otherwise ckpt hyperparameters are logged to MLflow, but yaml hyperparameters are used
+            )
+        else:
+            lightning_model = FasterRCNN(self.config)
+
+        # Get trainer
+        trainer = self.setup_trainer()  # this gets yaml!
```

Comment:
```suggestion
        trainer = self.setup_trainer()  
```
sorry sometimes I leave my random notes-to-self